temporal_sqlalchemy/bases.py

```diff
@@ -109,6 +115,10 @@ def record_history(self,
                 changes = attributes.get_history(clocked, prop.key)
 
             if changes.added:
+                if is_strict_mode:
+                    assert not is_vclock_unchanged, \
+                        'flush() has triggered for a changed temporalized property outside of a clock tick'
+
```

Comment:
we only increment the clock if a change is detected to the object, so it probably is safe being outside of the loop